addons/sale_timesheet/views/sale_timesheet_portal_templates.xml

```diff
@@ -3,7 +3,7 @@
 
     <template id="sale_order_portal_content_inherit" inherit_id="sale.sale_order_portal_template">
         <xpath expr="//li[.//a[@id='print_invoice_report']]" position="after">
-            <li t-if="sale_order.timesheet_count > 0 and sale_order.state in ('sale', 'done')" class="list-group-item flex-grow-1">
+            <li t-if="timesheets and sale_order.state in ('sale', 'done')" class="list-group-item flex-grow-1">
```

Comment:
It's too risky to make this change in the stable version. Instead, we should add a condition inside the View Timesheets button. It might leave a bit of empty space, but that should be acceptable.